dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchVectorStoreRecordCollection.cs

```diff
@@ -400,15 +400,15 @@ public async IAsyncEnumerable<TRecord> GetAsync(Expression<Func<TRecord, bool>>
             searchOptions.OrderBy.Add(name);
         }
 
-        VectorSearchResults<TRecord> vectorSearchResults = await this.SearchAndMapToDataModelAsync(null, searchOptions, options.IncludeVectors, cancellationToken).ConfigureAwait(false);
-        await foreach (var result in vectorSearchResults.Results.ConfigureAwait(false))
+        var vectorSearchResults = this.SearchAndMapToDataModelAsync(null, searchOptions, options.IncludeVectors, cancellationToken);
+        await foreach (var result in vectorSearchResults.ConfigureAwait(false))
```

Comment:
Since there's no await anywhere else in the method, isn't it possible to just return the result directly instead of doing a foreach?